### PR TITLE
[BEAM-6404] Fix issue with side inputs and flatten encoding.

### PR DESCRIPTION
**Pull request by robertwb**
The first commit is just a cleanup that turned out to not be essential to the fix, but was helpful in debugging and resolves a TODO.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @udim

File: sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py [hunk 1]
```diff
@@ -591,7 +591,6 @@ def sink_flattens(stages, pipeline_context):
         if pcollections[pcoll_in].coder_id != output_coder_id:
           # Flatten inputs must all be written with the same coder as is
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py]: I think this comment would be better like this:
`Flatten requires that all its inputs use the same coder as its output. Add stages to transcode Flatten inputs that do not use the same coder.`

Comment (robertwb) [reply sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py]: Thanks. Done.

**Comment by angoenka** [comment]
The PR break the build https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink_PR/133/console 

1st commit "Cleanup coder windowing." seems to be the cause.

**Comment by robertwb** [comment]
Hmm... I think that this is the new test that was always broken on Flink. I'll see how hard it is to roll forward, otherwise we can disable this in the meantime.

**Comment by mxm** [comment]
The test was passing before. Looks like it broken due to the coder changes in this PR.

**Comment by mxm** [comment]
```
ERROR: test_flattened_side_input (_main_.FlinkRunnerTest)
----------------------------------------------------------------------
Traceback (most recent call last)
File "apache_beam/runners/portability/fn_api_runner_test.py", line 205, in test_flattened_side_input
label='CheckFlattenOfSideInput')
File "apache_beam/pipeline.py", line 425, in _exit_
self.run().wait_until_finish()
File "apache_beam/runners/portability/portable_runner.py", line 349, in wait_until_finish
self._job_id, self._state, self._last_error_message()))
RuntimeError: Pipeline test_flattened_side_input_1547859357.36_07dcde9b-acfc-4e8d-b930-582f7637a07e failed in state FAILED: java.lang.IllegalArgumentException: PCollectionNodes [PCollectionNode
{id=ref_PCollection_PCollection_12, PCollection=unique_name: "17side3/Map(decode).None" coder_id: "ref_Coder_BytesCoder_1" is_bounded: BOUNDED windowing_strategy_id: "ref_Windowing_Windowing_1" }

] were consumed but never produced```

**Comment by mxm** [comment]
I see you modified the test. ValidatesRunner in PreCommit would have been useful here. Fine for me to disable the test if this is a limitation of the Flink side input, though I'm not sure it is.

**Comment by mxm** [comment]
Jira issue: https://issues.apache.org/jira/browse/BEAM-6473

**Pull request merged**